src/main/java/com/hillel/items_exchange/controller/UserController.java

```diff
@@ -0,0 +1,45 @@
+package com.hillel.items_exchange.controller;
+
+import com.hillel.items_exchange.dto.UserDto;
+import com.hillel.items_exchange.model.BaseEntity;
+import com.hillel.items_exchange.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.Principal;
+
+@RestController
+@RequestMapping("/user")
+@CrossOrigin(origins = "http://localhost:4200")
+@RequiredArgsConstructor
+@Slf4j
+public class UserController {
+
+    private final UserService userService;
+
+    @GetMapping("/info/{id}")
+    public @ResponseBody
+    ResponseEntity<UserDto> getUserInfo(@PathVariable("id") long id, Principal principal) {
+
+        userService.findByUsernameOrEmail(principal.getName())
+                .map(BaseEntity::getId)
+                .filter(userId -> userId == id)
+                .orElseThrow(() -> new AccessDeniedException("You do not have access to view the data of another user!"));
```

Comment:
Would you mind to extract message to property file?

Reply:
Done. Check it out, please.